diff --git a/ng1-narwhal.js b/ng2-narwhal.js
index 725036e..8024158 100644
--- a/ng1-narwhal.js
+++ b/ng2-narwhal.js
@@ -1,14 +1,14 @@
-myModule.component('myNarwhal', () => {
-    return {
-        bind: {
-            name: '@'
-        },
-        controller: Narhwal,
-        template: '<div>Hi {{ myNarwhal.bigName() }}</div>'
-    };
+@Component({
+    selector: 'my-narwhal',
+    bind: {
+        name: 'name'
+    }
});
+@Template({
+    inline: '<div>He is {{ bigName() }}.</div>'
+});
 class Narwhal {
     bigName() {
         return this.name.toUpperCase();
     }
 }
